k2/csrc/fsa_algo_test.cc

```diff
@@ -0,0 +1,79 @@
+// k2/csrc/fsa_algo_test.cc
+
+// Copyright (c)  2020  Fangjun Kuang (csukuangfj@gmail.com)
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/fsa_algo.h"
+
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace k2 {
+
+TEST(FsaAlgo, Connect) {
+  std::vector<Arc> arcs = {
+      {0, 1, 1}, {0, 2, 2}, {1, 3, 3}, {1, 6, 6},
+      {2, 4, 2}, {2, 6, 3}, {5, 0, 1},
+  };
+
+  std::vector<int32_t> arc_indexes = {0, 2, 4, 6, 6, 6, 7};
+
```

Comment:
maybe another case that: get an empty fsa after connecting a non-empty one

Reply:
We shouldn't need to specify the arc_indexes to the constructor of these functions because the begin-arc is part of the supplied arcs.  Would save coders time.  Fine for now though.